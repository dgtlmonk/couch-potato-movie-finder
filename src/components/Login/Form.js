import React from 'react'
import PropTypes from 'prop-types'

const Form = ({onSubmit}) => {
  return (
    <div>
      <form onSubmit={e => {
        const {username, password} = e.target.elements;
        e.preventDefault()
        onSubmit({
          username: username.value,
          password: password.value
        })
      }}>
        <label htmlFor="username">Username</label>
          <input data-testid="username" type="text" id="username"/>

        <label htmlFor="password">Password</label>
          <input data-testid="password" type="password" id="password" />

        <button type="submit">Login</button>
      </form>
    </div>
  )
}

Form.propTypes = {
  onSubmit : PropTypes.func.isRequired
}
export default Form
