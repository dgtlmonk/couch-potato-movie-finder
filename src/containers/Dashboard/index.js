import React from 'react'
import { Form as AntForm, Layout, Button, Input, Icon, Card, Row, Col } from 'antd'

const FormItem = AntForm.Item;
const {Header, Footer, Sider, Content} = Layout;

class Dashboard extends React.Component {
  render() {
    return(
      <div>
        <Layout>
          <Header>
            <div className="logo">Welcome Back</div>
            <div>
              <AntForm>
                <FormItem>
                  <Input data-testid="search"></Input>
                </FormItem>
              </AntForm>
            </div>
          </Header>
          <Content>
           <Row>
            <h3>Title</h3>
           </Row>
            <Row>
              <div data-testid="movie-list">
                <Col span={6}>
                </Col>
              </div>
            </Row>
          </Content>
        </Layout>
      </div>
    )
  }
}

export default Dashboard