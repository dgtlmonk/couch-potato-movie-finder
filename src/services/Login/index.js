const LoginSvc = {
requestLogin: async () => {
    return {
      status: 200,
      metadata: {
        message: 'Access Granted',
        user: 'chuck',
        permission: 100
      }
    }
  }
}

export default LoginSvc