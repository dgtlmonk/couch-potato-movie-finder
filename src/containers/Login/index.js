import React from 'react'
import {default as LoginForm } from 'components/Login/Form'
import { Card, Row, Col } from 'antd'
import PropTypes from 'prop-types'

class Login extends React.Component {
  static propTypes = {
    loginService: PropTypes.object.isRequired
  }

  static defaultProps = {
    loginService: undefined
  }

  onSubmit = ({ username, password }) => {
    this.props.loginService.requestLogin({ username, password })
  }

  render() {
    return (
    <div style={{ textAlign: "center",  padding: '30px' }} >
      <Row gutter={16}>
        <Col span={12} offset={6}>
          <Card style={{ width: 300, margin: 'auto' }}>
            <h2 data-testid='login-screen'>Login</h2>
            <LoginForm onSubmit={this.onSubmit}  />
          </Card>
        </Col>
      </Row>
    </div>
  )
  }
}

export default Login