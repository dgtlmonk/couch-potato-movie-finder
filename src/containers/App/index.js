import React from 'react'
import { Link, Route, Router, Switch } from 'react-router-dom'

import {createMemoryHistory} from 'history'
import styled from 'styled-components'
import {Layout} from 'antd'
import { default as LoginContainer } from 'containers/Login'
import Dashboard from 'containers/Dashboard'
import 'App.css';

const Wrapper = styled(Layout)`
  font-family: 'aktiv-grotesk-std', Helvetica Neue, Arial, sans-serif;
  background: #fff;
  padding: 0;
  ;`

const NoMatch = () => <div>No match</div>
const { Content } = Layout;

const LoginMockSvc  = {
  requestLogin: (...args) => {
    console.log(...args)
    return true
  }
}

class App extends React.Component {
  render() {
    return(
      <Wrapper>
        <Content>
          <Router>
          <Switch>
            <Route exact path="" render={routeProps => <LoginContainer {...routeProps} loginService={LoginMockSvc} />} />
            <Route path="/dashboard" component={Dashboard} />
            <Route component={NoMatch} />
          </Switch>
          {
            // <LoginContainer loginService={LoginMockSvc} />
          }
          </Router>
        </Content>
      </Wrapper>
    )
  }
}

export default App

