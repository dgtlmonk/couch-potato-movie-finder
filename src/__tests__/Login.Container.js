import React from 'react'
import 'jest-dom/extend-expect';
import {cleanup, renderIntoDocument} from 'react-testing-library'
import {default as Login} from '../containers/Login/';

// don't forget to clean up the document.body
afterEach(cleanup)

describe('<Login />', () => {
  let comp
  beforeEach(() => {
    comp = renderIntoDocument(<Login />)
  })

  describe('when rendered', () => {
    it('then should show login welcome text', () => {
       const { getByTestId } = comp

      expect(getByTestId('login-screen', { selector: 'h2' })).toBeInTheDOM()
    })
  });
})