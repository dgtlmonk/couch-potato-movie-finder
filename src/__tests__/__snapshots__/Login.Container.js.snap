// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> when rendered then should show login welcome text 1`] = `
Object {
  "container": <div>
    <div
        class="ant-layout"
    >
        <h2
            data-testid="login-screen"
        >
            Login
        </h2>
        <div>
            <form>
                <label
                    for="username"
                >
                    Username
                </label>
                <input
                    data-testid="username"
                    type="text"
                    id="username"
                />
                <label
                    for="password"
                >
                    Password
                </label>
                <input
                    data-testid="password"
                    type="password"
                    id="password"
                />
                <button
                    type="submit"
                >
                    Login
                </button>
            </form>
        </div>
        >
    </div>
</div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
